llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -23842,7 +23842,7 @@ SDValue DAGCombiner::visitCONCAT_VECTORS(SDNode *N) {
     }
 
     // concat_vectors(scalar, undef) -> scalar_to_vector(scalar)
-    if (!Scalar.getValueType().isVector()) {
+    if (!Scalar.getValueType().isVector() && In.hasOneUse()) {
```

Comment:
Scalar.hasOneUse? Also all of these combines seem to be checking one use, should the conditions be merged?

Reply:
It is the original In (not the Scalar after looking through the bitcast) that needs one use.